game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -72,6 +72,256 @@
+    private static final long serialVersionUID = -1015756641864182626L;
+  }
+
+  /**
+   * An action representing landing paratroopers.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class LandParatroopers implements IExecutable {
+    private static final long serialVersionUID = 5936225914851941086L;
+  }
+
+  /**
+   * An action representing marking no movement left.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class MarkNoMovementLeft implements IExecutable {
+    private static final long serialVersionUID = 1482181126769007815L;
+  }
+
+  /**
+   * An action representing attacker subs retreating before battle.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class AttackerRetreatSubsBeforeBattle implements IExecutable {
+    private static final long serialVersionUID = 2178507532676286044L;
+  }
+
+  /**
+   * An action representing definding subs retreating before battle.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class DefenderRetreatSubsBeforeBattle implements IExecutable {
+    private static final long serialVersionUID = -2081450648695833869L;
+  }
+
+  /**
+   * An action representing removing undefended transports.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class RemoveUndefendedTransports implements IExecutable {
+    private static final long serialVersionUID = 1369227461759133105L;
+  }
+
+  /**
+   * An action representing submerging subs vs only air.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class SubmergeSubsVsOnlyAir implements IExecutable {
+    private static final long serialVersionUID = -2588373599526477215L;
+  }
+
+  /**
+   * An action representing removing first strike suicide units.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class RemoveFirstStrikeSuicideUnits implements IExecutable {
+    private static final long serialVersionUID = 994041503171919922L;
+  }
+
+  /**
+   * An action representing standard attacking fire.
+   *
+   * <p>NOTE: This type exists solely for tests to interrogate the execution stack looking for an
+   * action of this type.
+   */
+  public abstract static class StandardAttackersFire implements IExecutable {
```

Comment:
If we later convert the battle steps to be something else, modules with injected functionality and step strings, we will need to remove these abstract classes. At the very least we will want to remove them so we avoid prod-code that is here only to support tests (a code smell).

Is there argument for not updating `IExecutable` to return a step-name enum value? This way if you have a list of `IExecutable`, you could call a `IExecutable#getStepName` method and check to see if it is an expected value, eg: `IExecutable.getStepName() == BattleStepName.STANDARD_ATTACKERS_FIRE`.

If you then restructure the steps to be individual classes and then compose them together to get the full battle sequence, I suspect any unit test checking the ordering of the steps would remain intact. WDYT?

Reply:
This naming of `IExecutables` was there originally for the sub first strike instances.  It is used by the RevisedTest and has been around since 2009.  I was just following that mechanism.

As for changing `IExecutables`, I don't like that idea.  They are used in a lot of places (battles, movements).  A grep shows 48 `new IExecutable` and another 7 places that implement `IExecutable` (ignoring the ones in `MustFightBattle`).  I'd have to edit a lot of places and I only need the step names in a few of the places.  So I'd either have to come up with a bunch of names for all of the places or put a placeholder in for the places that the tests don't care about.

I've pushed a change that deleted many of these test classes because they weren't being used for tests.  That leaves 12 of them, with two of them being around since 2009.